design/bridge_design.md

```diff
@@ -0,0 +1,317 @@
+  * The messaging and bridge are decentralized as much as possible.
+  * Block reorganizations should have no impact on the security of the messaging protocol and the downstream apps that use this protocol.
+
+
+2. **Bridge Requirements**
+  * The reference bridge contracts should be completely ordinary and separate from the management contract; The API should expose all the messaging bits required in order to build the bridge independently from Obscuro
+  * We want the bridge to have an asset whitelist controlled by the Obscuro DAO or simply configured by the management contract.
+  * Assets and their wrapped L2 counterparts should be mapped and exposed for querying.
+  * Bridge functionality should be able to use the [pull payment](https://docs.openzeppelin.com/contracts/2.x/api/payment#PullPayment) design. 
+   > If any current dApps want to extend or port their functionality on Obscuro then it should be possible for them to do it without collaborating with anyone else 
+
+## Scope
+* Messaging architecture between Obscuro and Mainnet Ethereum.
+* Basic Bridge that wraps assets
+* Primitive fees implementation
+
+## Assumptions
+
+1. Fees
+   * Big assumption - there will be a way to pay for bridge services/messaging in OBX without incurring losses in ETH on the layer 1 side.
+   * The Obscuro DAO will be able to configure properties for the messaging part of the protocol and accurately estimate the `gas costs` of storing data without losing too much on `gas price` spikes.
+2. Finality
+   * The management contract will be responsible for not exposing data that has not been finalized. This means that it should have first gone through the challenge period.
+   * The Rollup protocol will ensure the L1 blocks being consumed by the enclave and the resulting synthetic transactions are always bound to the blocks generating them.
+   * Obscuro has fast finality over the transaction ordering rather than the results. When a deposit gets reorganized the results on Obscuro L2 should reorganize too if depending on it. 
+
+
+
+## Modifications
+
+The Rollup headers right now are assumed to include withdrawal instructions. These instructions will change to generalized messages. Those messages might be withdrawal instructions or anything else.
+
+
+
+## MessageBus Design - Obscuro cross-chain data transmission 
+
+The `MessageBus` will be deployed as a [`Proxy`](https://docs.openzeppelin.com/contracts/4.x/api/proxy)
+
+The messaging API will be provided by the new `MessageBus` system contract. This contract's interface will be available for L1 and L2. This contract needs to be [`Ownable`](https://docs.openzeppelin.com/contracts/2.x/access-control) or [`RBAC`](https://docs.openzeppelin.com/contracts/2.x/access-control#role-based-access-control) based as some of the functions should only be callable from an administrative trusted address.
```

Comment:
let's define what an Obscuro "system contract" is somewhere in a "definitiions" section. It's the first time we introduce this concept